### DISCUSSION TITLE
Does agent check all classes when they are loaded into jvm?

### DISCUSSION THREAD

**Discussion by tuhao1020** [open-telemetry/opentelemetry-java-instrumentation#10071] [Q&A] [answered]
I have noticed that after agent injection, the time consumption increases for a certain method, which dynamically generates anonymous classes.
Does agent check all classes when they are loaded into jvm?
Is there a way to get around this? Like ignoring certain classes？

**Comment by laurit** [comment, marked as answer]
Agent needs to inspect all loaded classes to determine whether they need to be transformed, this can be expensive when you create a lot of class loaders (specially when you are running an old version as we have improved this in the last year). To ignore classes you can use https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/javaagent-extension-api/src/main/java/io/opentelemetry/javaagent/extension/ignore/IgnoredTypesConfigurer.java SPI in an agent extension. If you are creating a lot of class loaders then try https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/bdeec9fc0d478999ea4bfb62b3c32eef0064c8c1/javaagent-extension-api/src/main/java/io/opentelemetry/javaagent/extension/ignore/IgnoredTypesBuilder.java#L56
